packages/editor/src/editor.ts

```diff
@@ -127,7 +127,16 @@ export class Editor {
   }
 
   _isIdentifier(code: string): boolean {
-    const regex = /^([^!#%&()*+,\-/\\^`{|}~]+)$/;
+    if (this._isNumeric(code)) {
+      return false;
+    }
+
+    const regex = /^([^!"#%&'()*+,\-/\[\]\\^`{|}~]+)$/;
+    return regex.test(code);
+  }
+
+  _isNumeric(code: string) {
+    const regex = /^-?\d+$/;
```

Comment:
This matches things like `-42`, but not `-42.0` – I’m not sure if we need to be precise about this, we’re not going to be able to catch every case here with regexes anyway